src/interlocking_component/route_controller.py

```diff
@@ -330,6 +338,14 @@ def _get_interlocking_routes_for_edge(self, edge: Edge) -> List[Route]:
                 routes.append(route_candidate)
         return routes
 
-    def check_all_fahrstrassen_for_failures(self):
-        """This method checks for all trains, if their fahrstrassen and routes are still valid."""
-        raise NotImplementedError()
+    # def check_all_fahrstrassen_for_failures(self):
+    #    """This method checks for all trains, if their fahrstrassen and routes are still valid."""
+    #    raise NotImplementedError()
+
+    def recalculate_all_routes(self):
+        """Recalculates the route for every train in the simulation"""
+        self.routes_to_be_set = []
```

Comment:
If this get's merged after #438 yoe need to remember to also set routes_to_be_reserved to an empty List.

Reply:
I added a comment which can be commented in in this case